### PR TITLE
Specific parameters description

### PR DESCRIPTION
**Pull request by geofjamg**
Signed-off-by: Geoffroy Jamgotchian <geoffroy.jamgotchian@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the new behavior (if this is a feature change)?**
We can now better describe loadflow specific parameters, by creating a `Parameter` containing the name, the type, a description and list of possible values.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java [hunk 1]
```diff
@@ -115,49 +122,47 @@ public class OpenLoadFlowParameters extends AbstractExtension<LoadFlowParameters
...
+        new Parameter(VOLTAGE_REMOTE_CONTROL_PARAM_NAME, ParameterType.BOOLEAN, "Generator voltage remote control", VOLTAGE_REMOTE_CONTROL_DEFAULT_VALUE),
+        new Parameter(THROWS_EXCEPTION_IN_CASE_OF_SLACK_DISTRIBUTION_FAILURE_PARAM_NAME, ParameterType.BOOLEAN, "Throws an exception in case of slack distribution failure", THROWS_EXCEPTION_IN_CASE_OF_SLACK_DISTRIBUTION_FAILURE_DEFAULT_VALUE),
+        new Parameter(LOAD_POWER_FACTOR_CONSTANT_PARAM_NAME, ParameterType.BOOLEAN, "Load power factor is constant", LOAD_POWER_FACTOR_CONSTANT_DEFAULT_VALUE),
+        new Parameter(PLAUSIBLE_ACTIVE_POWER_LIMIT_PARAM_NAME, ParameterType.DOUBLE, "Plausible active power limit", LfNetworkParameters.PLAUSIBLE_ACTIVE_POWER_LIMIT_DEFAULT_VALUE),
+        new Parameter(SLACK_BUS_P_MAX_MISMATCH_NAME, ParameterType.DOUBLE, "Slack bus max active power mismatch", SLACK_BUS_P_MAX_MISMATCH_DEFAULT_VALUE),
+        new Parameter(VOLTAGE_PER_REACTIVE_POWER_CONTROL_NAME, ParameterType.BOOLEAN, "Reactive power / voltage slope", VOLTAGE_PER_REACTIVE_POWER_CONTROL_DEFAULT_VALUE),
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java:134]: The slope is in kV by MVar. So maybe change for description `Voltage per reactive power slope`

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java [hunk 2]
```diff
@@ -115,49 +122,47 @@ public class OpenLoadFlowParameters extends AbstractExtension<LoadFlowParameters
...
+        new Parameter(REACTIVE_POWER_REMOTE_CONTROL_PARAM_NAME, ParameterType.BOOLEAN, "SVC remote reactive power control", REACTIVE_POWER_REMOTE_CONTROL_DEFAULT_VALUE),
+        new Parameter(MAX_ITERATION_NAME, ParameterType.INTEGER, "Max iterations", NewtonRaphsonParameters.DEFAULT_MAX_ITERATION),
+        new Parameter(NEWTON_RAPHSON_CONV_EPS_PER_EQ_NAME, ParameterType.DOUBLE, "Newton-Raphson convergence epsilon per equation", DefaultNewtonRaphsonStoppingCriteria.DEFAULT_CONV_EPS_PER_EQ),
+        new Parameter(VOLTAGE_INIT_MODE_OVERRIDE_NAME, ParameterType.STRING, "Voltage init mode override", VOLTAGE_INIT_MODE_OVERRIDE_DEFAULT_VALUE.name(), Arrays.stream(VoltageInitModeOverride.values()).map(Enum::name).collect(Collectors.toList())),
+        new Parameter(TRANSFORMER_VOLTAGE_CONTROL_MODE_NAME, ParameterType.STRING, "Transformer voltage control mode", TRANSFORMER_VOLTAGE_CONTROL_MODE_DEFAULT_VALUE.name(), Arrays.stream(TransformerVoltageControlMode.values()).map(Enum::name).collect(Collectors.toList())),
+        new Parameter(DC_POWER_FACTOR_NAME, ParameterType.DOUBLE, "DC approx power factor", DC_POWER_FACTOR_DEFAULT_VALUE),
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java:140]: approx -> approximation ?

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java [hunk 3]
```diff
@@ -115,49 +122,47 @@ public class OpenLoadFlowParameters extends AbstractExtension<LoadFlowParameters
...
+        new Parameter(REACTIVE_RANGE_CHECK_MODE_NAME, ParameterType.STRING, "Reactive range check mode", LfNetworkParameters.REACTIVE_RANGE_CHECK_MODE_DEFAULT_VALUE.name(), Arrays.stream(ReactiveRangeCheckMode.values()).map(Enum::name).collect(Collectors.toList())),
+        new Parameter(LOW_IMPEDANCE_THRESHOLD_NAME, ParameterType.DOUBLE, "Low impedance threshold in per unit", LfNetworkParameters.LOW_IMPEDANCE_THRESHOLD_DEFAULT_VALUE),
+        new Parameter(NETWORK_CACHE_ENABLED_NAME, ParameterType.BOOLEAN, "Network cache enabled", NETWORK_CACHE_ENABLED_DEFAULT_VALUE),
+        new Parameter(SVC_VOLTAGE_MONITORING_NAME, ParameterType.BOOLEAN, "SVC voltage monitoring", SVC_VOLTAGE_MONITORING_DEFAULT_VALUE),
+        new Parameter(STATE_VECTOR_SCALING_MODE_NAME, ParameterType.STRING, "State vector scaling mode", NewtonRaphsonParameters.DEFAULT_STATE_VECTOR_SCALING_MODE.name(), Arrays.stream(StateVectorScalingMode.values()).map(Enum::name).collect(Collectors.toList())),
+        new Parameter(MAX_SLACK_BUS_COUNT_NAME, ParameterType.INTEGER, "Maximum slack bus count", LfNetworkParameters.DEFAULT_MAX_SLACK_BUS_COUNT)
```
Comment (annetill) [inline src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java:154]: Maximum slack buses count ?

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java (hunk 1 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java]: Done

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java (hunk 2 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java]: Done

File: src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java (hunk 3 above)
Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/OpenLoadFlowParameters.java]: Done

**Pull request merged**